cmd/qat_plugin/dpdkdrv/dpdkdrv_test.go

```diff
@@ -371,36 +365,39 @@ func TestScanPrivate(t *testing.T) {
 		},
 	}
 	for _, tt := range tcases {
-		if err := os.MkdirAll(tmpdir, 0750); err != nil {
-			t.Fatal(err)
-		}
+		t.Run(tt.name, func(t *testing.T) {
+			tmpdir, err := ioutil.TempDir("/tmp/", "qatplugin-TestScanPrivate-*")
+			if err != nil {
+				t.Fatal(err)
+			}
 
-		if err := createTestFiles(tmpdir, tt.dirs, tt.files, tt.symlinks); err != nil {
-			t.Fatalf("%+v", err)
-		}
+			if err = createTestFiles(tmpdir, tt.dirs, tt.files, tt.symlinks); err != nil {
+				t.Fatalf("%+v", err)
+			}
 
-		dp := &DevicePlugin{
-			maxDevices:      tt.maxDevNum,
-			pciDriverDir:    pciDrvDir,
-			pciDeviceDir:    pciDevDir,
-			dpdkDriver:      tt.dpdkDriver,
-			kernelVfDrivers: tt.kernelVfDrivers,
-		}
+			dp := &DevicePlugin{
+				maxDevices:      tt.maxDevNum,
+				pciDriverDir:    path.Join(tmpdir, "sys/bus/pci/drivers"),
+				pciDeviceDir:    path.Join(tmpdir, "sys/bus/pci/devices"),
+				dpdkDriver:      tt.dpdkDriver,
+				kernelVfDrivers: tt.kernelVfDrivers,
+			}
 
-		tree, err := dp.scan()
-		if tt.expectedErr && err == nil {
-			t.Errorf("Test case '%s': expected error, but got success", tt.name)
-		}
-		if !tt.expectedErr && err != nil {
-			t.Errorf("Test case '%s': got unexpected error: %+v", tt.name, err)
-		}
-		if len(tree) != tt.expectedDevNum {
-			t.Errorf("Test case '%s': expected %d, but got %d devices", tt.name, tt.expectedDevNum, len(tree))
-		}
+			tree, err := dp.scan()
+			if tt.expectedErr && err == nil {
+				t.Errorf("Test case '%s': expected error, but got success", tt.name)
+			}
+			if !tt.expectedErr && err != nil {
+				t.Errorf("Test case '%s': got unexpected error: %+v", tt.name, err)
+			}
+			if len(tree) != tt.expectedDevNum {
+				t.Errorf("Test case '%s': expected %d, but got %d devices", tt.name, tt.expectedDevNum, len(tree))
```

Comment:
Looks great! Still could you please drop these `Test case '%s':`  prefixes? They are not needed anymore since the cases are run through `t.Run()`.

Reply:
fixed, thanks!